qiita_pet/handlers/api_proxy/prep_template.py

```diff
@@ -6,52 +6,358 @@
+    """Get the summarized prep template data for each metadata column
+
+    Parameters
+    ----------
+    prep_id : int
+        PrepTemplate id to get info for
     user_id : str
-        User id requesting the prep templates
+        User requesting the sample template info
 
     Returns
     -------
-    dict
-        Prep template information in the form
-        {'status': status,
-         'message': msg,
-         'info': dict of list of dict}
-        status can be success, warning, or error depending on result
-        message has the warnings or errors
-        info contains prep template information seperated by data type, in the
-        form {data_type: [{prep 1 info dict}, ....], ...}
+    dict of objects
+        Dictionary object where the keys are the metadata categories
+        and the values are list of tuples. Each tuple is an observed value in
+        the category and the number of times its seen.
+        Format {status: status,
+                message: message,
+                num_samples: value,
+                category: [(val1, count1), (val2, count2), ...], ...}
+    """
+    prep = PrepTemplate(int(prep_id))
+    access_error = check_access(prep.study_id, user_id)
+    if access_error:
+        return access_error
+    df = prep.to_dataframe()
+    out = {'num_samples': df.shape[0],
+           'summary': {}}
+
+    # drop the prep_id column if it exists
+    if 'study_id' in df.columns:
+        df.drop('study_id', axis=1, inplace=True)
+    cols = list(df.columns)
+    for column in cols:
+        counts = df[column].value_counts()
+        out['summary'][str(column)] = [(str(key), counts[key])
+                                       for key in natsorted(counts.index)]
+    # Add expected messaging and status info
+    out.update({'status': 'success', 'message': ''})
+    return out
+
+
+@execute_as_transaction
+def prep_template_post_req(study_id, user_id, prep_template, data_type,
+                           investigation_type=None,
+                           user_defined_investigation_type=None,
+                           new_investigation_type=None):
+    """Adds a prep template to the system
+
+    Parameters
+    ----------
+    study_id : int
+        Study to attach the pre ptemplate to
+    user_id : str
+        User adding the prep template
+    prep_template : str
+        Filepath to the prep template being added
+    data_type : str
+        Data type of the processed samples
+    investigation_type: str, optional
+        Existing investigation type to attach to the prep template
+    user_defined_investigation_type: str, optional
+        Existing user added investigation type to attach to the prep template
+    new_investigation_type: str, optional
+        Investigation type to add to the system
     """
     access_error = check_access(study_id, user_id)
     if access_error:
         return access_error
-    # Can only pass ids over API, so need to instantiate object
```

Comment:
This code was moved to the study.py file in the api-proxy folder since it is a summary of what is in the study.